source/agora/api/FullNode.d

```diff
@@ -184,4 +185,24 @@ public interface API
 
     @method(HTTPMethod.GET)
     public bool hasEnrollment (Hash enroll_hash);
+
+    /***************************************************************************
+
+        Get validator's pre-image inforamtion at a certain height
+
+        API:
+            GET /get_preimage
+
+        Params:
+            enroll_key = The key for the enrollment in which the pre-image is
+                contained.
+            height = The block height of the preimage to get
+
+        Returns:
+            preimage information of the validator if exists, otherwise
+                PreimageInfo.init
+
+    ***************************************************************************/
+
+    public PreimageInfo getPreimage (Hash enroll_key, ulong height);
```

Comment:
Why do we pass `height` here ? The following would provide a much simpler API, wouldn't it ?

```suggestion
    public PreimageInfo getPreimage (Hash enroll_key);
```

Reply:
I agree with that. It would be simple.